src/interface/guide/components/Preparation/GemSubSection/GemBoxRow.tsx

```diff
@@ -0,0 +1,82 @@
+import styles from './GemBoxRow.module.scss';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import { CSSProperties, ReactNode } from 'react';
+import { Tooltip } from 'interface/index';
+import Item from 'common/ITEMS/Item';
+import Icon from 'interface/Icon';
+import { Gem as EventGem } from 'parser/core/Events';
```

Comment:
You are using two different `Item` types causing type errors. It looks like you want to use the one from `parser/core/Events` but you are importing the one from `common/ITEMS/Item` here.
```suggestion
import Icon from 'interface/Icon';
import { Gem as EventGem, Item } from 'parser/core/Events';
```

Reply:
That's on purpose.  More to do with the naming used in the codename itself.

There is the one from the event.  Then there is a common one and different parts pass something different.  I'm not at my computer but it was a bugger to track down and figure out.

Reply:
Are you certain?

Let's say `ItemA` is the one from `'parser/core/Events'` and `ItemB` is the one from `'common/ITEMS/Item'`

You are using the `ItemB` type for the `item` prop in the `GemBoxRowEntry` interface.
You then use `GemBoxRowEntry[]` as the return type for `getGemBoxRowEntries`.
You map the `GemBoxRowEntry[]` array from the record you get from `GemableGear`, which is typed `Record<number, ItemA>`.

```ts
get GemableGear(): Record<number, ItemA> {
  const gemSlots = this.GemableSlots;
  return Object.keys(gemSlots).reduce<Record<number, ItemA>>((obj, slot) => {
    const innerSlot = Number(slot);

    obj[innerSlot] = this.selectedCombatant._getGearItemBySlotId(innerSlot);

    return obj;
  }, {});
}

getGemBoxRowEntries(recommendedGems: Record<number, GemItem[]> = {}): GemBoxRowEntry[] {
  const gear: Record<number, ItemA> = this.GemableGear;
  // rest of code

  return Object.keys(gear)
  .filter((slot) => {
  })
  .map<GemBoxRowEntry>((slot) => {
    const item: ItemA = gear[slotNumber];
    // rest of code

    return {
      item, // ItemA - where ItemB is expected
      // rest of code
    };
  });
}
```